src/ZoneServer/Skills/Handlers/Swordsman/Hoplite/LongStride.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.Game.Const;
+using Melia.Shared.L10N;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters.Components;
+using Yggdrasil.Logging;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Swordsman.Hoplite
+{
+	/// <summary>
+	/// Handler for the Hoplite skill Long Stride
+	/// </summary>
+	[SkillHandler(SkillId.Hoplite_LongStride)]
+	public class Hoplite_LongStride : IGroundSkillHandler, IDynamicCasted
+	{
+		/// <summary>
+		/// Called when the user starts casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public async void StartDynamicCast(Skill skill, ICombatEntity caster)
+		{
+		}
+
+		/// <summary>
+		/// Called when the user stops casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void EndDynamicCast(Skill skill, ICombatEntity caster)
+		{
+		}
+
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!skill.Vars.TryGet<Position>("Melia.ToolGroundPos", out var targetPos))
+			{
+				caster.ServerMessage(Localization.Get("No target location specified."));
+				return;
+			}
+
+			if (!caster.Map.Ground.IsValidPosition(targetPos))
```

Comment:
Don't forget to add a check for collisions, because you were not able to jump over obstacles originally, an option to turn that off, as well as a check for the distance, to makes sure players don't jump wherever they want by modifying the client.

Reply:
Do we know you couldn't jump over obstacles originally?  I note that the "splash" part of the animation will display in the selected location even if it crosses out of bounds, so if you cut the jump short by checking for invalid terrain it messes up the animation.
I did add the max distance check, though.

Reply:
Actually, nevermind. I thought I had found proof that you couldn't jump over obstacles, but after doing some more research it actually seems like you _were_ able to. Honestly did not expect that^^